Added get related products by id endpoint

lj894 opened the pull request:
Added a route to support return related product ids group by relationship types.
The function name imported from model may subject to change after the actual function in model completed.  
Fixes #18

lj894 committed:
initial version of query-by-id

lj894 committed:
Query all related products by product id
Fixes #18

lj894 committed:
Update service.py

Added function to return all the related products and status for a given id. And added a helper class to support a custom return structure.

MinervaMint commented on service/service.py:
```diff
@@ -27,6 +37,57 @@ def index():
+            type_3_active.append(p.ref_id) if p.status else type_3_inactive.append(p.ref_id)
+
+    relationships.append(RelatedProducts(1, type_1_active, type_1_inactive))
+    relationships.append(RelatedProducts(1, type_1_active, type_1_inactive))
+    relationships.append(RelatedProducts(1, type_1_active, type_1_inactive))
```
Should these two lines be about another two types of relationships?

lj894 replied:
Fixed.

MinervaMint commented on service/service.py:
```diff
@@ -27,6 +37,57 @@ def index():
+    
+    # assume model returns recors in format of: [{id: 1, rel_id: 2, typeid: 1, status: true}]
+    relationships = []
+    type_1_active, type_1_inactive = [], []
```
Just for clarification, the relationship (`typeid`) are `0/1/2` right?

lj894 replied:
No from the story, the typeid is 1/2/3, and that's what the code being implemented. If we agree to use 0/1/2 instead, that'd be an easy fix on the code though.

MinervaMint replied:
Oh I just checked the story again and noticed that the `typeid` in the "assumption" part and in the "acceptance criteria" part are different - that's why I am confused 🤣 I guess we can all use 1/2/3.

lj894 replied:
Yeah that shouldn't be a big deal since the data will be created through our service anyway. As long as we follow the same standard from now on we should be fine. I thought start form 1 would be a good idea, since we are using integer id, I'm not sure if id 0 could be interpreted as boolean false.

MinervaMint replied:
Sure. I just changed the `typeid` range in my code to `1/2/3`.

MinervaMint commented on service/service.py:
```diff
@@ -27,6 +37,57 @@ def index():
+
+    for p in products:
+        if p.typeid == 1:
+            type_1_active.append(p.ref_id) if p.status else type_1_inactive.append(p.ref_id)
```
In the data schema (in model.py) the id of the recommended item is represented as `rel_id`.

lj894 replied:
Fixed.

lj894 committed:
Typo and bug fixes

Update the reference of ref_id to rel_id.
FIxed typo for adding relationship class to array.

MinervaMint commented on service/service.py:
```diff
@@ -27,6 +37,57 @@ def index():
+
+    relationships.append(RelatedProducts(1, type_1_active, type_1_inactive))
+    relationships.append(RelatedProducts(2, type_2_active, type_2_inactive))
+    relationships.append(RelatedProducts(1, type_3_active, type_3_inactive))
```
I guess this line should be `relationships.append(RelatedProducts(3, type_3_active, type_3_inactive))`?

MinervaMint replied:
I mean line 88

lj894 replied:
Yes you are right. It's crazy I changed all the code in my IDE and when I use github desktop to push the update, it did not push my latest code. More like the code from my last save. How did you guys push your update each time? Are you using command line or directly update the file from github site?

lj894 replied:
These last 3 lines in my local file was correct since yesterday but every time I push an update it missed something. So weird. I wonder if I have some set up messed up. Just like last time I literally have an extra line in my local file, but it didn't show up here on the website. I had to put 2 extra lines locally so that the it shows up 1 here...

MinervaMint replied:
That is indeed weird 😂... I am using command line.

lj894 replied:
Ok thank you so much. I will make sure to keep an eye on my code next time after I push just in case.

lj894 committed:
Typo fixed

Fixed typo for adding relationship class.
